### PR TITLE
Allow branch workspace

### PR DESCRIPTION
**Pull request by ErikZigo**
Jira: https://keboola.atlassian.net/browse/KBC-1478
KBC: https://github.com/keboola/connection/pull/3188

Before asking for review make sure that:

- [x] ~~New client method(s) has tests~~ Nové metody v clientovi nejsou
- [x] Apiary file is updated
- [x] You declared if there is a BC break or not (will affect next release of a client)

---

### CODE REVIEW

File: src/Keboola/StorageApi/BranchAwareClient.php [hunk 1]
```diff
@@ -14,7 +14,10 @@ public function __construct($branch, array $config = [])
 
     public function request($method, $url, $options = array(), $responseFileName = null, $handleAsyncTask = true)
     {
-        $url = 'branch/' . $this->branch . '/' . $url;
+        if (strpos($url, 'jobs/') !== 0) {
+            $url = 'branch/' . $this->branch . '/' . $url;
+        }
```
Comment (ErikZigo) [inline src/Keboola/StorageApi/BranchAwareClient.php:19]: Joby nemame branchovany, nefungovalo by autoaticky handlovani async odpovedi.

File: tests/Common/BranchComponentTest.php [hunk 2]
```diff
@@ -1463,21 +1463,4 @@ public function testConfigurationRollback()
             'currentVersion',
         ]);
     }
-
-    /**
-     * @param string $branchPrefix
-     */
-    protected function deleteBranchesByPrefix(DevBranches $devBranches, $branchPrefix)
```
Comment (ErikZigo) [inline tests/Common/BranchComponentTest.php:1470]: Neni potreba, vyresi BranchWorkspacesTest

File: tests/Common/DevBranchesTest.php [hunk 3]
```diff
@@ -296,38 +288,6 @@ public function testReadOnlyRoleBranchesPermissions()
         $this->assertCount(2, $adminDevBranches->listBranches());
     }
 
-    public function testCreateAndDeleteBranchWithWorkspace()
```
Comment (ErikZigo) [inline tests/Common/DevBranchesTest.php:299]: Neni potreba, to se odbavuje v `BranchWorkspacesTest` / `WorkspacesTest`

File: tests/Common/BranchComponentTest.php (hunk 2 above)
Comment (tomasfejfar) [reply tests/Common/BranchComponentTest.php:1470]: Klidně to mohl být trait ale whatever.

**Review by tomasfejfar — APPROVED** [review]
Přidal jsem test na to, že se nemíchají workspacy z brancí do listování workspaců jinak OK

**Pull request merged**